src/ceph-volume/ceph_volume/decorators.py

```diff
@@ -11,7 +11,7 @@ def needs_root(func):
     """
     @wraps(func)
     def is_root(*a, **kw):
-        if not os.getuid() == 0:
+        if not os.getuid() == 0 and not os.environ.get('CEPH_VOLUME_SKIP_NEEDS_ROOT', False):
```

Comment:
When exactly do we need root and when not? I mean we certainly can't create OSDs without being root, right?

This feels as if ceph-volume itself should know when it requires root.

Reply:
I mentioned it in the tracker but this is needed when running ceph-volume in a root-less container where the UID is not 0 (in Rook's case it's ceph UID 167). In this scenario, the container can still be privileged or run with the right Kernel capabilities.

Reply:
TIL!.